frontend/src/pages/Notifications/helper.js

```diff
@@ -1,29 +1,91 @@
+  let subprojectDisplayName = "";
+
   if (metadata !== undefined) {
-    if (metadata.workflowitem !== undefined && metadata.workflowitem.hasViewPermissions === true) {
-      return metadata.workflowitem.displayName;
-    }
-    if (metadata.subproject !== undefined && metadata.subproject.hasViewPermissions === true) {
-      return metadata.subproject.displayName;
+    if (metadata.project) {
+      projectDisplayName = isAllowedToSeeType(metadata, project) ? getDisplayName(metadata, project) : redacted;
+      if (metadata.subproject) {
+        subprojectDisplayName = isAllowedToSeeType(metadata, subproject)
+          ? getDisplayName(metadata, subproject)
+          : redacted;
+      }
     }
-    if (metadata.project !== undefined && metadata.project.hasViewPermissions === true) {
-      return metadata.project.displayName;
+  }
+
+  return {
+    projectDisplayName,
+    subprojectDisplayName
+  };
+}
+
+export const getType = notificationMetaData => {
+  if (notificationMetaData.project !== undefined) {
+    if (notificationMetaData.subproject !== undefined) {
+      if (notificationMetaData.workflowitem !== undefined) {
+        return workflowitem;
+      }
+      return subproject;
     }
+    return project;
+  }
+  return null;
+};
+
+export function getDisplayName(notificationMetaData, type) {
+  if (type === project) {
+    return notificationMetaData.project.displayName;
+  }
+  if (type === subproject) {
+    return notificationMetaData.subproject.displayName;
+  }
+  if (type === workflowitem) {
+    return notificationMetaData.workflowitem.displayName;
   }
   return "";
 }
 
+const isAllowedToSeeType = (notificationMetaData, type) => {
+  if (type === project) {
+    return notificationMetaData.project.hasViewPermissions;
+  }
+  if (type === subproject) {
+    return notificationMetaData.subproject.hasViewPermissions;
+  }
+  if (type === workflowitem) {
+    return notificationMetaData.workflowitem.hasViewPermissions;
+  }
+};
+
 export const intentMapping = notification => {
   const businessEvent = notification.businessEvent;
+  if (!businessEvent) {
+    console.warn("Notification has no business event");
+    return "";
+  }
   const translation = strings.notification[businessEvent.type];
-  const displayName = getDisplayName(notification);
-  const text = formatString(translation, displayName);
-  return `${text} ${isAllowedToSee(notification) ? "" : strings.notification.no_permissions}`;
+
+  const notificationMetaData = notification.metadata;
+  if (!notificationMetaData) {
+    console.warn("Notification has no metadata");
+    return "";
+  }
+
+  const type = getType(notificationMetaData);
+  const displayName = getDisplayName(notificationMetaData, type);
+  const text = isAllowedToSeeType(notificationMetaData, type)
+    ? formatString(translation, displayName)
+    : formatString(translation, "");
+  return `${text} ${isAllowedToSeeType(notificationMetaData, type) ? "" : strings.notification.no_permissions}`;
```

Comment:
You are evaluating the same condition twice

Reply:
moved addition of `strings.notification.no_permissions` into line 87